pkcs8/src/private_key_info.rs

```diff
@@ -167,23 +184,26 @@ impl<'a> PrivateKeyInfo<'a> {
     }
 }
 
-impl<'a> DecodeValue<'a> for PrivateKeyInfo<'a> {
+impl<'a, Params, Key> DecodeValue<'a> for PrivateKeyInfoInner<Params, Key>
+where
+    Params: der::Choice<'a, Error = der::Error> + Encode,
+    Key: From<&'a [u8]>,
+{
     type Error = der::Error;
 
-    fn decode_value<R: Reader<'a>>(
-        reader: &mut R,
-        header: Header,
-    ) -> der::Result<PrivateKeyInfo<'a>> {
+    fn decode_value<R: Reader<'a>>(reader: &mut R, header: Header) -> der::Result<Self> {
         reader.read_nested(header.length, |reader| {
             // Parse and validate `version` INTEGER.
             let version = Version::decode(reader)?;
             let algorithm = reader.decode()?;
-            let private_key = OctetStringRef::decode(reader)?.into();
+            let private_key: &[u8] = OctetStringRef::decode(reader)?.into();
+            let private_key = Key::try_from(private_key)?;
```

Comment:
yeah, this could be an OctetString or OctetStringRef instead. I don't know what I was thinking last year.

Reply:
`clippy` seems to suggest `From` over `TryFrom`

Reply:
Yeah, clippy lead me to rethink my position on OctetString actually.
I rewrote the backing storage in the follow up commit.